### ISSUE TITLE
Support Eastern Arabic numerals for lists

### ISSUE THREAD

**Issue by mustafa0x** [jgm/pandoc#6725]
```
$ pandoc -v
pandoc 2.10.1

$ pandoc
1. hello
2. world
<ol type="1">
<li>hello</li>
<li>world</li>
</ol>

$ pandoc
١. مرحبا
٢. بالعالم
<p>١. مرحبا ٢. بالعالم</p>
```

Perhaps something like this could be supported? A quick test with babelmark indicated that https://github.com/dotnet/docfx supports this.

**Comment by mb21** [comment]
if you're talking about HTML output, you can use some CSS to make that happen...

**Comment by mustafa0x** [comment]
Thanks! That's definitely an option, but being forced to use Western Arabic numerals (i.e., 1, 2, 3, ...) is a problem for a variety of reasons which I discuss here:

https://github.com/micromark/micromark/issues/26#issuecomment-704770434

If this can even be supported by a flag that'd be a big improvement. As things are, pre-processing would be needed to convert Eastern Arabic to Western Arabic.

**Comment by jgm** [comment]
In general I think it's a strength of markdown that it doesn't force you to write formatting commands in a foreign language.  We might want to consider localization of numerals at some point, but it does open up a lot of complexities.

Note that pandoc's markdown allows you to create an ordered list this way, without using numerals at all:
```
#. one
#. two
```
Maybe that would be helpful?

**Comment by alerque** [comment]
> a strength of markdown that it doesn't force you to write formatting commands in a foreign language

If your language is not one that uses (Western) Arabic numerals then they *are* the foreign language. One of the strengths of Markdown is that it's human readable¹. This doesn't extend to list numbering which forces a different language on what is arguable **content**, not 'formatting commands'.

All Unicode characters already have character class encodings that identify their functions. You don't have to explicitly support Arabic for this, you can catch a whole host of languages quite easily. You just need to match on all things that are Unicode numbers followed by a period (Perl RegEx = `\l{N}+\.`) rather than only ASCI numbers `[0-9]+\.`.

If this definition of valid list numbering has not made it into CommonMark I suggest that it should! And yes I think it should make it into Pandoc flavored markup whether CM adopts it or not.

¹ <sub>Another place this comes up is lazy numbering. I like writing lists with `1. ... 1. ...` (or I guess `#. ...`) and letting the rendering step take care of the counting. This does make some editing operations easier (no renumbering to re-arrange) but it also makes it less human readable. You can see at a glance that it is a numbered list but you can't also see at a glance what the counter is for a given item, you have to count.</sub>

**Comment by jgm** [comment]
> If your language is not one that uses (Western) Arabic numerals then they are the foreign language. One of the strengths of Markdown is that it's human readable¹. This doesn't extend to list numbering which forces a different language on what is arguable content, not 'formatting commands'.

Yes, this is what I was saying.  (I don't think we disagree.)

**Comment by alerque** [comment]
> Yes, this is what I was saying. (I don't think we disagree.)

Okay I'm sorry, I interpreted quite the opposite out of that statement, but I do see now how we could be meaning the same thing.

**Comment by jgm** [comment]
> just need to match on all things that are Unicode numbers followed by a period (Perl RegEx = \l{N}+\.) rather than only ASCI numbers [0-9]+\..

I don't think it's this simple.  For one thing, this may catch too many things.  For another thing, we need to know not just whether the string is numeric, but what the number is (for the start number).  And the same unicode characters may correspond to different numbers in different languages, for all I know. (Then there are additional issues when you start to consider lettered enumerated lists.)

That's why this feature would require localization.  You'd need to specify the language being parsed, so we could choose the right characters accordingly.

**Comment by mustafa0x** [comment]
Thanks for the replies!

Yeah, specifying a locale seems seems like a safe approach.

**Comment by alerque** [comment]
> lettered enumerated lists

Okay *that* really is a can of worms. (Also maybe a reason I could never make it in academia or government.)

**Comment by jgm** [comment]
For example, we've had a request before to support Spanish lettered list numbering with includes `ñ`.

**Comment by alerque** [comment]
Yes, I get where this is going. Even more conniptions would be required for Turkish, a language I speak and work with a lot. Besides the extra letters sandwiched in among your basic English alphabet it also has a couple of omissions. There is also no consistency as to which of these variations to consider when using letters for lists. Some (but not all) people skip the additions, others use them (A→B→C→Ç→D...). What to do with `ı` is a perennial problem. Even more confusingly the same people that skip over Turkish additions to the English subset of the Latin alphabet may or may not skip over the omissions (i.e. they may or may not use `Q`, `X`, or `W`).

**Comment by ekm507** [comment]
I wanted to remind you that number localization is not implemented yet :)

some programming languages like python, support numbers from different languages out of the box. eg:
```python
int('۱۲۳۴۵۶۷۸۹۰')
# output: 1234567890
```
but I don't know if something like this works in Haskell or not.
thanks.
